### PR TITLE
Template linting for TPv2

### PR DESCRIPTION
**Pull request by ocket8888**
This adds some linting rules for templates in TPv2. Right now, it's spitting out a bunch of warnings, which might be disabled entirely in the future (each one has an accompanying rationale in the configuration file). It also makes the code compliant with the linting rules. The biggest change is just that all buttons now must have an explicit `type` attribute - even when using the implicit default `submit` type. That helps catch buttons that *shouldn't* be `submit` type, but also buttons incorrectly used as links etc.

<hr/>

## Which Traffic Control components are affected by this PR?
- Traffic Portal (experimental v2)

## What is the best way to verify this PR?
Make sure the code passes the new linting rules (albeit with warnings)

## PR submission checklist
- [x] This PR has tests
- [x] This PR doesn't need documentation
- [x] This PR doesn't need a CHANGELOG.md entry
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

**Review by rimashah25 — CHANGES_REQUESTED** [review]
The code change looks good to me. 
I tried to test the changes on local to ensure buttons and anchors are still doing what they are supposed to and I am unable to login since login button is not working. So, I can't test any other buttons.

File: experimental/traffic-portal/src/app/login/reset-password-dialog/reset-password-dialog.component.html [hunk 1]
```diff
@@ -21,6 +21,6 @@ <h1 mat-dialog-title>Reset Password</h1>
 	</mat-dialog-content>
 	<mat-dialog-actions>
 		<button mat-raised-button color="warn" type="button" mat-dialog-close>Cancel</button>
-		<button mat-raised-button color="primary">Submit</button>
+		<button  mat-raised-button color="primary" type="submit">Submit</button>
```
[the remark this answers is not in the document]
Comment (ocket8888) [reply experimental/traffic-portal/src/app/login/reset-password-dialog/reset-password-dialog.component.html:24]: mistake

File: experimental/traffic-portal/src/app/core/servers/server-details/server-details.component.html [hunk 2]
```diff
@@ -189,7 +189,7 @@
 				</div>
 			</fieldset>
 			<div class="buttons">
```
Comment (rimashah25) [inline experimental/traffic-portal/src/app/core/servers/server-details/server-details.component.html:191]: submit button isn't working.

File: experimental/traffic-portal/src/app/core/servers/phys-loc/table/phys-loc-table.component.html [hunk 3]
```diff
@@ -26,4 +26,4 @@
 	</tp-generic-table>
 </mat-card>
```
Comment (rimashah25) [inline experimental/traffic-portal/src/app/core/servers/phys-loc/table/phys-loc-table.component.html:28]: there isn't a link on physical location page to create division.

**Review by rimashah25 — CHANGES_REQUESTED** [review]

File: experimental/traffic-portal/src/app/core/servers/server-details/server-details.component.html (hunk 2 above)
Comment (ocket8888) [reply experimental/traffic-portal/src/app/core/servers/server-details/server-details.component.html:191]: Submit button _"works"_ just fine, problem is the controller has no handler for editing servers. It only creates them; which is a bug not caused by this PR.

Comment (rimashah25) [reply experimental/traffic-portal/src/app/core/servers/server-details/server-details.component.html:191]: Ok, creating an [issue](https://github.com/apache/trafficcontrol/issues/7485) for the same.

**Review by rimashah25 — APPROVED** [review]
LGTM. Tested locally button/link changes for login/users/physicalLocations/server/cachegroup. Unable to test deliveryservice or anything related to it.

**Pull request merged**